src/main/java/com/example/mybookshopapp2/model/Author.java

```diff
@@ -0,0 +1,48 @@
+package com.example.mybookshopapp2.model;
+
+import com.fasterxml.jackson.annotation.JsonIdentityInfo;
+import com.fasterxml.jackson.annotation.ObjectIdGenerators;
+import io.swagger.annotations.ApiModel;
+import io.swagger.annotations.ApiModelProperty;
+import lombok.Data;
+import lombok.EqualsAndHashCode;
+import org.springframework.hateoas.RepresentationModel;
+
+import javax.persistence.*;
+import java.util.HashSet;
+import java.util.Set;
+
+@Data
+@EqualsAndHashCode(exclude = "books")
+@Entity
+@Table(name = "authors")
+@ApiModel(description = "data model of author entity")
+@JsonIdentityInfo(
```

Comment:
It's not good to use the same class object for the db model and the response in controllers. It's better to use DTO classes for the controllers. You should not mix persistence annotations with json and swagger annotations.